src/braket/pennylane_plugin/translation.py

```diff
@@ -25,18 +26,71 @@
     StateVector,
     Variance,
 )
+from braket.schema_common import BraketSchemaBase
 from pennylane.operation import Observable, ObservableReturnTypes, Operation
 
 from braket.pennylane_plugin.ops import PSWAP, XY, YY, CPhaseShift00, CPhaseShift01, CPhaseShift10
 
+_BRAKET_TO_PENNYLANE_OPERATIONS = {
+    "x": "PauliX",
+    "y": "PauliY",
+    "z": "PauliZ",
+    "h": "Hadamard",
+    "ry": "RY",
+    "rx": "RX",
+    "rz": "RZ",
+    "s": "S",
+    "vi": "SX",
+    "t": "T",
+    "cnot": "CNOT",
+    "cy": "CY",
+    "cz": "CZ",
+    "swap": "SWAP",
+    "cswap": "CSWAP",
+    "ccnot": "Toffoli",
+    "phaseshift": "PhaseShift",
+    "unitary": "QubitUnitary",
+    "amplitude_damping": "AmplitudeDamping",
+    "generalized_amplitude_damping": "GeneralizedAmplitudeDamping",
+    "phase_damping": "PhaseDamping",
+    "depolarizing_channel": "DepolarizingChannel",
+    "bit_flip": "BitFlip",
+    "phase_flip": "PhaseFlip",
+    "qubit_channel": "QubitChannel",
+    "cphaseshift": "ControlledPhaseShift",
+    "cphaseshift00": "CPhaseShift00",
+    "cphaseshift01": "CPhaseShift01",
+    "cphaseshift10": "CPhaseShift10",
+    "iswap": "ISWAP",
+    "pswap": "PSWAP",
+    "xy": "XY",
+    "xx": "IsingXX",
+    "yy": "YY",
+    "zz": "IsingZZ",
+}
+
+
+def supported_operations(device) -> FrozenSet[str]:
+    """Returns the operations supported by the plugin based upon the device.
 
-def supported_operations() -> FrozenSet[str]:
-    """Returns the operations supported by the plugin.
+    Args:
+        device (Device): The device to obtain the supported operations for
 
     Returns:
         FrozenSet[str]: The names of the supported operations
     """
-    return frozenset(op.__name__ for op in _translate_operation.registry) - {"object"}
+    braket_ops = _BRAKET_TO_PENNYLANE_OPERATIONS
+    supported_ops = frozenset()
+    try:
+        device_list = device.properties.action["braket.ir.jaqcd.program"].supportedOperations
+        device_list = [x.lower() for x in device_list]
+        supported_ops = frozenset(device_list)
+    except AttributeError:
+        raise AttributeError("Device needs to have properties defined.")
+    translated_supported_ops = {
+        str(braket_ops[op]) for op in braket_ops if len(supported_ops) > 0 and str(op).lower() in supported_ops
```

Comment:
No need for the `str` casts, nor the `len(supported_ops) > 0` check. Also, just call `frozenset` directly on the comprehension and return it:
```python
return frozenset(_BRAKET_TO_PENNYLANE_OPERATIONS[op] for op in _BRAKET_TO_PENNYLANE_OPERATIONS if op.lower() in supported_ops)
```